nixos/modules/services/backup/borgbackup.nix

```diff
@@ -284,7 +284,7 @@ in {
           };
 
           archiveBaseName = mkOption {
-            type = types.strMatching "[^/{}]+";
+            type = types.nullOr (types.strMatching "[^/{}]+");
```

Comment:
`prune.prefix` is just a `str`, so this may still cause type errors. disabling pruning when `prune.prefix == null` makes sense, but that should be reflected in the types and descriptions of pruning options as well.